python/ray/serve/_private/proxy_response_generator.py

```diff
@@ -56,6 +57,13 @@ def stop_checking_for_disconnect(self):
         self._disconnected_task = None
 
 
+def swallow_cancelled(task: asyncio.Task):
+    try:
+        task.result()
+    except (RequestCancelledError, asyncio.CancelledError):
+        pass
+
```

Comment:
the reason why we are getting `RequestCancelledError` is because there is a race condition between `response.cancel` and exception stemming from trying to access a cancelled future when there are desco errors. we should try to figure out the cause of that race condition and make it deterministic instead of swallowing the error

Reply:
synced offline, we will go with this but add a comment explaining the reasoning